src/modules/weather.js

```diff
@@ -0,0 +1,102 @@
+const weatherModule = () => {
+  const locationInput = document.getElementById('location-input');
+  const submitButton = document.getElementById('submit-weather-form');
+  const nameElemenet = document.getElementById('name');
+  const tempElement = document.getElementById('temp');
+  const weatherElement = document.getElementById('weather');
+  const humidityElement = document.getElementById('humidity');
+  const dataContainer = document.getElementById('data-container');
+  const loader = document.getElementById('loader-container');
+  const tempToggle = document.getElementById('temp-toggle');
+
+
+  const fetchWeather = async (location, unit = false) => {
+    const unitValue = unit === true ? 'metric' : 'imperial';
+    const response = await fetch(`https://api.openweathermap.org/data/2.5/weather?q=${location}&units=${unitValue}&appid=4bd4b9ab3b3c5a1d664f8bc0eecfaf36`, { mode: 'cors' });
+    loader.style.display = 'flex';
+    const weatherData = await response.json();
```

Comment:
Keep all the styling logic in CSS files (If you need to change something like a background image, add the classes with the different backgrounds to the CSS and then change the class name attribute with JavaScript) ([screenshot](https://gitlab.com/microverse/guides/projects/requirements_screenshots/raw/master/images/javascript/section/styling_requirement.png) from Pathwright)